src/data/finsupp/fin.lean

```diff
@@ -0,0 +1,99 @@
+/-
+Copyright (c) 2021 Ivan Sadofschi Costa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Ivan Sadofschi Costa
+-/
+import data.fin.tuple
+import data.finsupp.basic
+
+/-! `cons` and `tail` for maps `fin n →₀ M` and their properties -/
+
+noncomputable theory
+
+namespace finsupp
+
+variable {n : ℕ}
+
+/-- `tail` for maps `fin (n+1) →₀ M`. See `fin.tail` for more details. -/
```

Comment:
Here you say "for maps `fin (n+1) →₀ M`" but you have `fin (n+1) →₀ ℕ`. You should generalize everything similarly to [fin.tail](https://leanprover-community.github.io/mathlib_docs/data/fin/tuple/basic.html#fin.tail).